src/ixdat/readers/zilien.py

```diff
@@ -1,61 +1,314 @@
+
+        Args:
+            data (numpy.array): The data block of the tsv file as an array
+            metadata (dict): Extracted metadata
+            timestamp (float): The timestamp of the measurement
+            series_headers (List[str]): List of series headers, slots with empty strings
+                means the same as the last non-empty one
+            column_headers (List[str]): List of column headers
+
+        Returns:
+            List[Series], DefaultDict(str, List[str]): List of series and aliases
+        """
+        series = []
+        last_time_series = None
+        aliases = defaultdict(list)
+        last_series_header = ""
+
+        # Iterate over numbered series and column headers
+        for column_number, (series_header, column_header) in enumerate(
+            zip(series_headers, column_headers)
+        ):
+            last_series_header = series_header or last_series_header
+
+            # Skip series not relevant for the type of measurement
+            if not issubclass(self._cls, ECMeasurement) and last_series_header == "pot":
+                continue
+            elif not issubclass(self._cls, MSMeasurement) and to_mass(
+                last_series_header
+            ):
+                continue
+
+            # Pluck column of the correct length out from the data block and form series
+            count = metadata[f"{last_series_header}_{last_series_header}_count"]
+            column_data = data[:count, column_number]
+
+            # Form the series_name, unit, aliases and update aliases
+            series_name, unit, standard_name = self._form_names_and_unit(
+                last_series_header, column_header
+            )
+            if standard_name:
+                aliases[standard_name].append(series_name)
+
+            # Form series kwargs and the series
+            series_kwargs = {
+                "name": series_name,
+                "unit_name": unit,
+                "data": column_data,
+            }
+            if column_header == "Time [s]":  # Form TimeSeries
+                column_series = TimeSeries(**series_kwargs, tstamp=timestamp)
+                last_time_series = column_series
+            else:
+                column_series = ValueSeries(**series_kwargs, tseries=last_time_series)
+
+            series.append(column_series)
+
+        return series, aliases
+
+    @staticmethod
+    def _form_names_and_unit(series_header, column_header):
+        """Return names and unit
+
+        Args:
+            series_header (str): Something like "Iongauge value" or "C0M18"
+            column_header (str): Something like "Time [s]" or "Time [s]"
+
+        Returns:
+            str, str, Optional[str]: Return series_name, unit, standard_name
+
+        """
+        standard_name = None
+        if column_header == "Time [s]":  # Form TimeSeries
+            unit = "s"
+            if series_header == "pot":
+                name = f"Potential {column_header.lower()}"
```

Comment:
They could. But since they will no depend on the content of the file, I think I would prefer to keep the there for now. Will change that, when there is a zilien file format version 2